pkg/features/table.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"fmt"
+)
+
+// Table provides a structure for table-driven tests.
+// Each entry in the table represents an executable assessment.
+type Table []struct {
+	Name       string
+	Assessment Func
+}
+
+// Build converts the defined test steps in the table
+// into a FeatureBuilder which can be used to add additional attributes
+// to the feature before it's exercised.
+func (table Table) Build(name string) *FeatureBuilder {
```

Comment:
Can name be empty?

Reply:
Looks like it is used here: https://github.com/kubernetes-sigs/e2e-framework/blob/84123945caf59b33f30ce1cd8f1d9e01a0d3c93b/pkg/env/env.go#L310

and using an empty string does not provide what I was looking for (which is avoiding this sub-test layer with this extra name)

Reply:
I debated around this myself. In a table-driven setup, the top test should be named automatically.
The issue with the extra-subtest layer is another one that I want to address eventually (#42).
Let me take another pass at it and see what I come up with.